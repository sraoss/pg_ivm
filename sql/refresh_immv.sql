CREATE TABLE t (i int PRIMARY KEY);
INSERT INTO t SELECT generate_series(1, 5);

SELECT create_immv('mv', 'SELECT * FROM t');

SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
SELECT refresh_immv('mv', false);
SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;

INSERT INTO t VALUES(6);
SELECT i FROM mv ORDER BY 1;

SELECT refresh_immv('mv', true);
SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
SELECT i FROM mv ORDER BY 1;

INSERT INTO t VALUES(7);
SELECT i FROM mv ORDER BY 1;

SELECT refresh_immv('mv', false);
SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
SELECT i FROM mv ORDER BY 1;

INSERT INTO t VALUES(8);
SELECT i FROM mv ORDER BY 1;


