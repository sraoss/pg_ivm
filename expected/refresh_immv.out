CREATE TABLE t (i int PRIMARY KEY);
INSERT INTO t SELECT generate_series(1, 5);
SELECT create_immv('mv', 'SELECT * FROM t');
NOTICE:  created index "mv_index" on immv "mv"
 create_immv 
-------------
           5
(1 row)

SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
 immvrelid | withnodata 
-----------+------------
 mv        | f
(1 row)

SELECT refresh_immv('mv', false);
 refresh_immv 
--------------
            5
(1 row)

SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
 immvrelid | withnodata 
-----------+------------
 mv        | f
(1 row)

INSERT INTO t VALUES(6);
SELECT i FROM mv ORDER BY 1;
 i 
---
 1
 2
 3
 4
 5
 6
(6 rows)

SELECT refresh_immv('mv', true);
 refresh_immv 
--------------
            0
(1 row)

SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
 immvrelid | withnodata 
-----------+------------
 mv        | t
(1 row)

SELECT i FROM mv ORDER BY 1;
 i 
---
(0 rows)

INSERT INTO t VALUES(7);
SELECT i FROM mv ORDER BY 1;
 i 
---
(0 rows)

SELECT refresh_immv('mv', false);
 refresh_immv 
--------------
            7
(1 row)

SELECT immvrelid, withnodata FROM pg_ivm_immv ORDER BY 1;
 immvrelid | withnodata 
-----------+------------
 mv        | f
(1 row)

SELECT i FROM mv ORDER BY 1;
 i 
---
 1
 2
 3
 4
 5
 6
 7
(7 rows)

INSERT INTO t VALUES(8);
SELECT i FROM mv ORDER BY 1;
 i 
---
 1
 2
 3
 4
 5
 6
 7
 8
(8 rows)

